### ISSUE TITLE
openntpd -s flag deprecated: how can I force synchronization ?

### ISSUE THREAD

**Issue by teto** [NixOS/nixpkgs#111899]
**Describe the bug**
I have a dual boot windows/linux. Booting windows messes up my clock in linux by 1 hour (I have ` time.hardwareClockInLocalTime` to false, I know I can fix it by configuring windows to save it in non localtime too but for the sake of the argument, stay with me. I used to run `sudo ntpd -s` to force a resync but that's apparently a thing of the past:
```
$ sudo ntpd -s
-s option no longer works and will be removed soon.
Please reconfigure to use constraints or trusted servers.
```
I've added constraints and restarted it
```
cat /etc/ntpd.conf                                                                                                                                                                        ~
server 0.nixos.pool.ntp.org
server 1.nixos.pool.ntp.org
server 2.nixos.pool.ntp.org
server 3.nixos.pool.ntp.org
constraints from "https://www.google.com/"
constraints from "https://www.nixos.org/"
```
with this log
```
Feb 04 13:07:38 jedha systemd[1]: Starting OpenNTP Server...
Feb 04 13:07:38 jedha ntpd[15232]: ntp engine ready
Feb 04 13:07:39 jedha ntpd[15232]: constraint reply from 216.58.214.68: offset -3596.721515
Feb 04 13:07:39 jedha ntpd[15232]: constraint reply from 2a00:1450:4007:806::2004: offset -3596.731402
Feb 04 13:07:39 jedha ntpd[15232]: cancel settime because offset is negative or close enough
Feb 04 13:07:39 jedha systemd[1]: Started OpenNTP Server.
Feb 04 13:07:39 jedha ntpd[15232]: constraint reply from 2a03:b0c0:3:d0::d24:5001: offset -3596.855468
Feb 04 13:07:39 jedha ntpd[15232]: constraint reply from 165.22.65.139: offset -3596.855715
Feb 04 13:07:39 jedha ntpd[15232]: constraint reply from 134.209.226.211: offset -3596.861501
Feb 04 13:07:39 jedha ntpd[15232]: constraint reply from 2a05:d014:275:cb00:29f:95e8:f0de:2bdd: offset -3596.873497
```

I've tried "trusting" a server
```
server 1.nixos.pool.ntp.org
server 2.nixos.pool.ntp.org
server 0.nixos.pool.ntp.org trusted
constraints from "https://www.google.com/"
constraints from "https://www.nixos.org/
```
to no avail
```
Feb 04 13:38:18 jedha systemd[1]: Stopped OpenNTP Server.
Feb 04 13:38:18 jedha systemd[1]: openntpd.service: Consumed 150ms CPU time, received 44.7K IP traffic, sent 24.5K IP traffic.
Feb 04 13:38:18 jedha systemd[1]: Starting OpenNTP Server...
Feb 04 13:38:18 jedha ntpd[23113]: ntp engine ready
Feb 04 13:38:19 jedha ntpd[23113]: cancel settime because offset is negative or close enough
Feb 04 13:38:19 jedha systemd[1]: Started OpenNTP Server.
Feb 04 13:38:19 jedha ntpd[23113]: constraint reply from 2a00:1450:4007:80f::2004: offset -3596.577563
Feb 04 13:38:19 jedha ntpd[23113]: constraint reply from 216.58.198.196: offset -3596.581681
Feb 04 13:38:19 jedha ntpd[23113]: constraint reply from 18.195.76.55: offset -3596.728738
Feb 04 13:38:19 jedha ntpd[23113]: constraint reply from 2a03:b0c0:3:d0::d19:7001: offset -3596.730574
Feb 04 13:38:19 jedha ntpd[23113]: constraint reply from 18.198.68.141: offset -3596.735960
Feb 04 13:38:19 jedha ntpd[23113]: constraint reply from 2a03:b0c0:3:e0::1b:1: offset -3596.801684
Feb 04 13:38:37 jedha ntpd[23113]: peer 37.59.38.13 now valid
Feb 04 13:38:40 jedha ntpd[23113]: peer 2001:bc8:2a01:100::42 now valid
Feb 04 13:38:40 jedha ntpd[23113]: peer 37.187.104.44 now valid
Feb 04 13:40:04 jedha ntpd[23231]: adjusting local clock by -3595.753295s
```
even though it says `adjusting local clock by -3595.753295s`, a call to  `Thu Feb  4 01:42:56 PM CET 2021` is still 1 hour more.

@thoughtpolice asking here since I was unsure if https://github.com/openntpd-portable/openntpd-portable/issues was the correct tracker to ask for help ?

One may want to change the example for
```
    extraOptions = mkOption {
      type = with types; separatedString " ";
      default = "";
      example = "-s";
      description = ''
        Extra options used when launching openntpd.
      '';
```
sinmce `-s` is deprecated.

**Notify maintainers**

this is on unstable

**Comment by teto** [comment]
actually my hardware clock assumes it's UTC when it is set to CET time instead
```
sudo hwclock --show --verbose
[sudo] password for teto: 
hwclock from util-linux 2.36.1
System Time: 1612443215.577741
Trying to open: /dev/rtc0
Using the rtc interface to the clock.
Last drift adjustment done at 1612426153 seconds after 1969
Last calibration done at 1612426153 seconds after 1969
Hardware clock is on UTC time
Assuming hardware clock is kept in UTC time.
Waiting for clock tick...
...got clock tick
Time read from Hardware Clock: 2021/02/04 12:53:40
Hw clock time : 2021/02/04 12:53:40 = 1612443220 seconds since 1969
Time since last adjustment is 17067 seconds
Calculated Hardware Clock drift is 0.000000 seconds
2021-02-04 13:53:39.655853+01:00
```

**Comment by teto** [comment]
I've switched to services.ntpd instead

**Comment by thoughtpolice** [comment]
@teto Sorry for the delay on this. Yes, the documentation should certainly be fixed at the very least, but more broadly I've actually thought of deprecating and removing openntpd anyway, for reasons like this one (the inability to immediately force a synchronization and wait on that to occur also makes `openttpd` an unusable solution for daemons like Ceph or CockroachDB that mandate precise synchronization before startup; letting clock sync occur in the background asynchronously isn't acceptable and it's not possible to paper over this discrepancy in any way, AFAICS.) I think the reality is just that the openntpd project is tailored exclusively for the needs and desires of OpenBSD developers, and its usage on Linux is only "nice to have" due to the work of volunteers making it possible, but it's otherwise not a development priority. So stuff like this (behavioral differences, deprecation of `-s` out of nowhere, etc) is bound to happen due to cultural differences.

Again, sorry for the delay here.

**Comment by teto** [comment]
thanks for the insight ! No issue if you looked at timestamps i opend and closed the issue in 20mn; it served more like a ruibberduck thing xD
